storage/innobase/row/row0ins.cc

```diff
@@ -1568,6 +1573,74 @@ class ib_dec_in_dtor {
 	ulint&		counter;
 };
 
+/*********************************************************************//**
+Reads sys_trx_end field from clustered index row.
+@return trx_id_t */
+static
+trx_id_t
+row_ins_get_sys_trx_end(
+/*===================================*/
+			const rec_t *rec,	/*!< in: clustered row */
+			ulint *offsets,		/*!< in: offsets */
+			dict_index_t *index)	/*!< in: clustered index */
+{
+	ut_a(dict_index_is_clust(index));
+
+	ulint len;
+	ulint nfield = dict_col_get_clust_pos(
+		&index->table->cols[index->table->vers_end], index);
+	const byte *field = rec_get_nth_field(rec, offsets, nfield, &len);
+	ut_a(len == 8);
+	return(mach_read_from_8(field));
+}
+
+/**
+Performs search at clustered index and returns sys_trx_end if row was found.
+@param[in]	index	secondary index of record
+@param[in]	rec	record in a secondary index
+@param[out]	end_trx_id	value from clustered index
+@return DB_SUCCESS, DB_NO_REFERENCED_ROW */
+static
+dberr_t
+row_ins_search_sys_trx_end(
+	dict_index_t *index,
+	const rec_t *rec,
+	trx_id_t *end_trx_id)
```

Comment:
Can we return `trx_id_t`, and `@retval 0 if the row was not found`?
I would prefer to avoid output parameters.

When the parameter list fits in a single 79-column line, then write it in a single line.
Else, align also the parameter names with TABs.

In InnoDB, the `&` or `*` must be attached to the type name, not to the variable name.